ucte/ucte-converter/src/main/java/com/powsybl/ucte/converter/UcteImporter.java

```diff
@@ -927,4 +924,71 @@ public Network importData(ReadOnlyDataSource dataSource, NetworkFactory networkF
         }
     }
 
+    @Override
+    public boolean exists(ReadOnlyDataStore dataStore, String fileName) {
+        try {
+            UcteDataFormat df = new UcteDataFormat();
```

Comment:
UcteDataFormat, XmlDataFormat and so one should be singleton. Maybe you should convert these class to enums with a single entry `INSTANCE`